libtenzir/src/metrics_collector.cpp

```diff
@@ -70,15 +63,12 @@ auto metrics_collector(
   self->send(self, atom::run_v);
   return {
     [self](atom::run) -> caf::result<void> {
-      // Build and import stream on first usage.
-      auto stream = import_stream::make(self, self->state.node);
-      if (!stream) {
-        TENZIR_WARN("{} aborts because it failed to create import stream: {}",
-                    *self, stream.error());
-        return stream.error();
-      }
-      self->state.importer
-        = std::make_unique<import_stream>(std::move(*stream));
+      auto scoped_self = caf::scoped_actor{self->system()};
+      auto components
+        = get_node_components<importer_actor>(scoped_self, self->state.node);
+      TENZIR_ASSERT_CHEAP(components);
+      auto [importer] = std::move(*components);
+      self->state.importer = std::move(importer);
```

Comment:
We can get the importer directly from the node state when we spawn the metrics collector actor in `make_component` here: https://github.com/tenzir/tenzir/blob/4ebca1698b1b3ed4374b85ca639aecde38c0c5cf/libtenzir/builtins/components/metrics_collector.cpp#L44-L47